cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsRequestParser.java

```diff
@@ -414,17 +411,18 @@ private static void parseTimeseries(Map<String, List<String>> queryParams, Metri
     if (queryParams.containsKey(START_TIME) && queryParams.containsKey(END_TIME)) {
       startTime = TimeMathParser.parseTime(now, queryParams.get(START_TIME).get(0));
       endTime = TimeMathParser.parseTime(now, queryParams.get(END_TIME).get(0));
-      if (!queryParams.containsKey(RESOLUTION)) {
-        // determine resolution, based on difference.
-        MetricsRequest.TimeSeriesResolution autoResolution = getResolution(endTime - startTime);
-        builder.setTimeSeriesResolution(autoResolution.getResolution());
-        resolution = autoResolution.getResolution();
-      }
-      if (queryParams.containsKey(COUNT)) {
-        count = Integer.parseInt(queryParams.get(COUNT).get(0));
+      if (queryParams.containsKey(RESOLUTION)) {
+        if (queryParams.get(RESOLUTION).get(0).equals("auto")) {
+          //auto determine resolution, based on time difference.
+          MetricsRequest.TimeSeriesResolution autoResolution = getResolution(endTime - startTime);
+          builder.setTimeSeriesResolution(autoResolution.getResolution());
+          resolution = autoResolution.getResolution();
+        }
       } else {
-        count = (int) (((endTime / resolution * resolution) - (startTime / resolution * resolution)) / resolution + 1);
+        builder.setTimeSeriesResolution(MetricsRequest.TimeSeriesResolution.SECOND.getResolution());
```

Comment:
init resolution variable before this line and use it in builder.setTimeSeriesResolution(resolution) - will make code more readable